tests/hydra.rs

```diff
@@ -440,3 +452,209 @@ fn scenario_2() -> TestResult {
     ];
     test_scenario(payloads, "tests/hydra/scenario_2.txt")
 }
+
+#[test]
+fn hydra_emulation_scenario_1() -> TestResult {
+    hydra_emulation_test("tests/hydra/scenario_1.txt".to_string())
+}
+
+#[test]
+fn hydra_emulation_scenario_2() -> TestResult {
+    hydra_emulation_test("tests/hydra/scenario_2.txt".to_string())
+}
+
+#[test]
+fn hydra_oura_stdout_scenario_1() -> TestResult {
+    let tx_event = r#"{"event":"apply","point":{"hash":"84e657e3dd5241caac75b749195f78684023583736cc08b2896290ab","slot":7},"record":{"hex":"84a300d9010281825820f0a39560ea80ccc68e8dffb6a4a077c8927811f06c5d9058d0fa2d1a8d047d2000018282581d605e4e214a6addd337126b3a61faad5dfe1e4f14f637a8969e3a05eefd1a001e848082581d600d45f2b310a98e766cee2ab2f6756c91719bd7b35929cef058365b651a015ef3c00200a100d90102818258200f193a88190f6dace0a3db1e0e50797a6e28cd4b6e289260dc96b5a8d7934bf858401b13ee550f3167a1b94796f2a2f5e22d782d628336a7797c5b798f358fa564dbe92ea75a4e2449eb2cef59c097d8497545ef1e4ea441b88a481194323ae7c608f5f6"}}"#;
+    let events = vec![
+        "PeerConnected".to_string(),
+        "PeerConnected".to_string(),
+        "Greetings".to_string(),
+        "HeadIsInitializing".to_string(),
+        "Committed".to_string(),
+        "Committed".to_string(),
+        "Committed".to_string(),
+        "HeadIsOpen".to_string(),
+        "TxValid".to_string(),
+        tx_event.to_string(),
+        "SnapshotConfirmed".to_string(),
+        "HeadIsClosed".to_string(),
+        "ReadyToFanout".to_string(),
+        "HeadIsFinalized".to_string(),
+    ];
+    hydra_oura_stdout_test("tests/hydra/scenario_1.txt".to_string(), events)
```

Comment:
I tried mutating the expectations: tweaking 1) tx binary, 2) tx event slot, 3) dropping an event, but the test appears to still pass

Reply:
now it signals when failing. nice catch.thanks